db/backup.go

```diff
@@ -0,0 +1,266 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package db
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/datastore"
+)
+
+func (db *db) basicImport(ctx context.Context, txn datastore.Txn, filepath string) (err error) {
+	f, err := os.Open(filepath)
+	if err != nil {
+		return NewErrOpenFile(err)
+	}
+	defer func() {
+		closeErr := f.Close()
+		if closeErr != nil {
+			err = NewErrCloseFile(closeErr, err)
+		}
+	}()
+
+	d := json.NewDecoder(bufio.NewReader(f))
+
+	t, err := d.Token()
+	if err != nil {
+		return err
+	}
+	if t != json.Delim('{') {
+		return ErrExpectedJSONObject
+	}
+	for d.More() {
+		t, err := d.Token()
+		if err != nil {
+			return err
+		}
+		colName := t.(string)
+		col, err := db.getCollectionByName(ctx, txn, colName)
```

Comment:
todo: These are not scoped to a the input txn param, and as such, actions on them, such as `Create` will not be scoped to the provided transaction either.  Please scope these to the txn.

Reply:
✅